contracts/rust/adapter/src/sol_types.rs

```diff
@@ -312,289 +313,375 @@ impl PartialEq for ConsensusKeysUpdatedV2 {
     }
 }
 
-impl Serialize for ValidatorRegistered {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: Serializer,
-    {
-        (&self.account, &self.blsVk, &self.schnorrVk, self.commission).serialize(serializer)
-    }
-}
-
-#[allow(non_snake_case)]
-impl<'de> Deserialize<'de> for ValidatorRegistered {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
-    where
-        D: Deserializer<'de>,
-    {
-        let (account, blsVk, schnorrVk, commission) = <(_, _, _, u16)>::deserialize(deserializer)?;
-        Ok(Self {
-            account,
-            blsVk,
-            schnorrVk,
-            commission,
-        })
-    }
-}
-
-impl Serialize for ValidatorRegisteredV2 {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: Serializer,
-    {
-        (
-            &self.account,
-            &self.blsVK,
-            &self.schnorrVK,
-            self.commission,
-            &self.blsSig,
-            &self.schnorrSig,
-        )
-            .serialize(serializer)
-    }
-}
-
-#[allow(non_snake_case)]
-impl<'de> Deserialize<'de> for ValidatorRegisteredV2 {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
-    where
-        D: Deserializer<'de>,
-    {
-        let (account, blsVK, schnorrVK, commission, blsSig, schnorrSig) =
-            <(_, _, _, u16, _, _)>::deserialize(deserializer)?;
-        Ok(ValidatorRegisteredV2 {
-            account,
-            blsVK,
-            schnorrVK,
-            commission,
-            blsSig,
-            schnorrSig,
-        })
-    }
-}
-
-impl Serialize for CommissionUpdated {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: Serializer,
-    {
-        (
-            &self.validator,
-            self.timestamp,
-            self.oldCommission,
-            self.newCommission,
-        )
-            .serialize(serializer)
+#[derive()]
+/**Event with signature `ValidatorRegistered(address,(uint256,uint256,uint256,uint256),(uint256,uint256),uint16)` and selector `0xf6e8359c57520b469634736bfc3bb7ec5cbd1a0bd28b10a8275793bb730b797f`.
+```solidity
+event ValidatorRegistered(address indexed account, BN254.G2Point blsVk, EdOnBN254.EdOnBN254Point schnorrVk, uint16 commission);
+```*/
+#[allow(
+    non_camel_case_types,
+    non_snake_case,
+    clippy::pub_underscore_fields,
+    clippy::style
+)]
+#[derive(Clone, Serialize, Deserialize)]
+pub struct ValidatorRegisteredLegacy {
+    #[allow(missing_docs)]
```

Comment:
rather than manually replicating these, can we use [`sol!` to generate](https://docs.rs/alloy/latest/alloy/sol_types/macro.sol.html) instead? 

we can define like:

```rust
sol! {
  struct ValidatorRegsiteredLegacy {}
}
```

all the serde should also be auto-derived, if not then use `#[sol(all_derives)]` on the struct

Reply:
dunno how to apply this, directly copy your code above doesn't work

Reply:
Would these derives be compatible with our manual derives? The reason for keeping the manual derives is because we need to migrate existing storage to the new auto derived serialization format.